setup.cfg

```diff
@@ -0,0 +1,54 @@
+[metadata]
+name = pyfscacher
+description = Caching results of operations on heavy file trees
+long_description = file:README.md
+long_description_content_type = text/markdown
+author = Center for Open Neuroscience
+author_email = debian@onerussian.com
+maintainer = John T. Wodder II
+maintainer_email = pyfscacher@varonathe.org
+license = MIT
+license_files = LICENSE
+url = https://github.com/con/pyfscacher
+
+keywords =
+    ###
+
+classifiers =
+    Development Status :: 3 - Alpha
+    #Development Status :: 4 - Beta
+    #Development Status :: 5 - Production/Stable
+    Programming Language :: Python :: 3 :: Only
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: Implementation :: CPython
+    Programming Language :: Python :: Implementation :: PyPy
+    License :: OSI Approved :: MIT License
+    ###
+
+project_urls =
+    Source Code = https://github.com/con/pyfscacher
+    Bug Tracker = https://github.com/con/pyfscacher/issues
+
+[options]
+packages = find:
+package_dir =
+    =src
+python_requires = ~=3.6
+install_requires =
+    appdirs == 1.*
+    joblib >= 0.17
+
+[options.packages.find]
+where = src
```

Comment:
Having a `src/` directory ensures that testing always happens against an installed copy of the code instead of the local copy, thereby catching errors in packaging.  It also helps avoid common mistakes where non-package `.py` files are bundled or where package data files aren't bundled.  Other arguments: [[1]](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure) [[2]](https://hynek.me/articles/testing-packaging/)

I don't see how it complicates editable installs.  The code within the `src/` directory will still be added to `sys.path`.  In fact, editable installs will be cleaner, as code outside `src/` (like `setup.py`) won't be added to `sys.path`.

Reply:
Thanks. ok, let's try going this way then and see what it brings (joy or pain ;))